portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -50,56 +70,39 @@ private static String convertToJsonUsingGson(Map<String, String> oldFacts) {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form.
-    String userFirstName = getParameter(request, /*content=*/ FIRST_NAME_KEY, /*defaultValue=*/ "");
-    String userLastName = getParameter(request, /*content=*/ LAST_NAME_KEY, /*defaultValue=*/ "");
+    String userName = getParameter(request, /*content=*/ USER_NAME_KEY, /*defaultValue=*/ "");
     String userFunFact = getParameter(request, /*content=*/ FUN_FACT_KEY, /*defaultValue=*/ "");
 
-    // No need to change oldFacts if there is no new fact
+    // No need to add to datastore if there is no new fact
     if (userFunFact.isEmpty()) {
       response.sendRedirect("/index.html");
       return;
     }
 
-    // Define key and value pair to add to oldFacts
-    String factKey = getFactKey(userFirstName, userLastName);
-    String factValue = getFactValue(oldFacts, userFunFact, factKey);
-    
-    // Store user's input to access later
-    oldFacts.put(factKey, factValue);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Define key to add to datastore
+    String factKey = getFactKey(userName);
+
+    Entity funFactEntity = new Entity(ENTITY_NAME);
+    funFactEntity.setProperty(PROPERTY_NAME, factKey);
+    funFactEntity.setProperty(PROPERTY_FACT, userFunFact);
+    datastore.put(funFactEntity);
 
-    // Redirect back to the HTML page.
     response.sendRedirect("/index.html");
-    response.getWriter().println(factKey + ": " + factValue);
   }
 
-  private static String getFactKey(String firstName, String lastName) {
+  private static String getFactKey(String userName) {
     String factKey;
-    if (firstName.isEmpty() && lastName.isEmpty()) {
+    if (userName.isEmpty()) {
       factKey = NO_NAME;
     }
-    else if (firstName.isEmpty()) {
-      factKey = lastName;
-    }
-    else if (lastName.isEmpty()) {
-      factKey = firstName;
-    }
     else {
-      factKey = firstName + " " + lastName;
+      factKey = userName;
     }
     return factKey;
```

Comment:
nit: since this is a lot simpler now, it's slightly simpler to just return early for each if statement so you don't need to define factKey. So something like:

if (userName.isEmpty()) {
  return NO_NAME;
}

return userName;

In fact, this would be good even with the multiple conditions shown before. But not a big deal, just slightly fewer lines and easier to read.

Reply:
I see, I actually just switched it to using the ternary conditional operator since it reduces to one line. Let me know if you'd like me to switch it back to a separate method though!